From 2de0c6011b5a400ec38672bef96a7dd55f5d36dd Mon Sep 17 00:00:00 2001
From: Nathan Poirier <nathan@poirier.io>
Date: Sat, 28 Jul 2018 21:57:16 +0200
Subject: [PATCH 1/3] Ignore css resources

The files are already versioned by the css plugin. A new versionning will breaks the css files.
---
 index.js | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/index.js b/index.js
index c43ddc3..b989607 100644
--- a/index.js
+++ b/index.js
@@ -7,6 +7,9 @@ class LaravelMixFilenameVersioning {
       const newAssets = {};
 
       Object.keys(stats.compilation.assets).forEach(assetName => {
+        if (assetName.indexOf('?') !== -1) { // ignore css resources
+          return;
+        }
         let originalAssetNameParts = path.parse(assetName);
         let newAssetFile = new File(path.join(Config.publicPath, assetName));
         let newAssetFileName = newAssetFile.segments.name + '.' + newAssetFile.version().substr(0, 8) + newAssetFile.segments.ext;
@@ -38,4 +41,4 @@ class LaravelMixFilenameVersioning {
   }
 }
 
-module.exports = LaravelMixFilenameVersioning;
\ No newline at end of file
+module.exports = LaravelMixFilenameVersioning;

From 16977018abff02a7a88c3e5c316b50165da974b5 Mon Sep 17 00:00:00 2001
From: Nathan Poirier <nathan@poirier.io>
Date: Sun, 29 Jul 2018 09:32:43 +0200
Subject: [PATCH 2/3] Skip files based on Config.fileLoaderDirs paths

---
 index.js | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/index.js b/index.js
index b989607..d2e602d 100644
--- a/index.js
+++ b/index.js
@@ -7,8 +7,10 @@ class LaravelMixFilenameVersioning {
       const newAssets = {};
 
       Object.keys(stats.compilation.assets).forEach(assetName => {
-        if (assetName.indexOf('?') !== -1) { // ignore css resources
-          return;
+        for (let path of Config.fileLoaderDirs) {
+          if (assetName.indexOf(path + '/') !== -1) {
+            return;
+          }
         }
         let originalAssetNameParts = path.parse(assetName);
         let newAssetFile = new File(path.join(Config.publicPath, assetName));

From f1c9ead1ea947302d5b621f79a000e7e25aee5f3 Mon Sep 17 00:00:00 2001
From: Nathan Poirier <nathan@poirier.io>
Date: Sun, 29 Jul 2018 09:34:31 +0200
Subject: [PATCH 3/3] Fixup

---
 index.js | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/index.js b/index.js
index d2e602d..11dbced 100644
--- a/index.js
+++ b/index.js
@@ -7,8 +7,8 @@ class LaravelMixFilenameVersioning {
       const newAssets = {};
 
       Object.keys(stats.compilation.assets).forEach(assetName => {
-        for (let path of Config.fileLoaderDirs) {
-          if (assetName.indexOf(path + '/') !== -1) {
+        for (let key in Config.fileLoaderDirs) {
+          if (Config.fileLoaderDirs.hasOwnProperty(key) && assetName.indexOf(Config.fileLoaderDirs[key] + '/') !== -1) {
             return;
           }
         }
